### PR TITLE
AlertingNG: remove warn/crit from eval prototype

### PR DESCRIPTION
**Pull request by kylebrandt**
and misc cleanup

want to drop the warn/crit from the design. While it can be useful, it also complicates things. The complications span across all the components that make up the alerting logic so I think it will make the behavior hard to implement and hard for users to always follow.
Also, when thinking about making alerting so it can integrate with different IRMs, having more particular notions in the system makes it more likely to not fit with IRMs in a logical way.

**Special notes for your reviewer**:
This is a prototype behind a feature flag

### CODE REVIEW

File: pkg/services/ngalert/eval.go [hunk 1]
```diff
@@ -176,14 +184,14 @@ func (ng *AlertNG) LoadAlertConditions(dashboardID int64, panelID int64, conditi
...
-	return &conditions, nil
+	condition.RefID = conditionRefID
+	return &condition, nil
 }
 
-// Execute runs the WarnCondition and CritCondtion expressions or queries.
-func (conditions *Conditions) Execute(ctx AlertExecCtx, fromStr, toStr string) (*ExecutionResult, error) {
-	result := ExecutionResult{}
-	if !conditions.IsValid() {
+// Execute runs Conditions' expressions or queries.
```
Comment (kylebrandt) [inline pkg/services/ngalert/eval.go:191]:
```suggestion
// Execute runs the Condition's expressions or queries.
```

**Review by papagian — APPROVED** [review]
Nice!

File: pkg/services/ngalert/eval.go [hunk 2]
```diff
@@ -33,66 +36,71 @@ func init() {
...
 	Ctx context.Context
 }
 
-// At least Warn or Crit condition must be non-empty
-type Conditions struct {
-	Condition string `json:"condition"`
+// Condition contains backend expressions and queries and the RefID
+// of the query or expression that will be evaluated.
+type Condition struct {
+	RefID string `json:"condition"`
```
[the remark this answers is not in the document]
Comment (kylebrandt) [reply pkg/services/ngalert/eval.go:54]: nope oversight

Comment (kylebrandt) [inline pkg/services/ngalert/eval.go:54]:
```suggestion
	RefID string `json:"refId"`
```

File: pkg/services/ngalert/eval.go [hunk 3]
```diff
@@ -33,66 +36,71 @@ func init() {
...
 
-type EvalResult struct {
+// Result contains the evaluated state of an alert instance
+// identified by its labels.
+type Result struct {
 	Instance data.Labels
 	State    State // Enum
 }
 
+// State is a enum of the evaluation state for an alert instance.
```
Comment (aknuds1) [inline pkg/services/ngalert/eval.go:79]:
```suggestion
// State is an enum of the evaluation state for an alert instance.
```

**Review by aknuds1 — APPROVED** [review]
Looks good AFAICT. You go Kyle.

**Pull request merged**